pkg/iac/scanners/helm/parser/parser.go

```diff
@@ -95,16 +96,16 @@ func (p *Parser) ParseFS(ctx context.Context, target fs.FS, path string) error {
 			return nil
 		}
 
-		if detection.IsArchive(path) {
-			tarFS, err := p.addTarToFS(path)
+		if detection.IsArchive(filePath) && !isDependencyChartArchive(p.workingFS, filePath) {
```

Comment:
i think this will only avoid the problem where we skip (ie, do not unpack) helm charts, but will still fail (with the `file does not exist` error) for the same reasons described in the issue #7781 in other cases where we do not skip - eg:
```
tree backup/
backup/
├── Chart.yaml
├── common-2.26.0.tgz
└── opentelemetry-collector-0.108.0.tgz
```
`trivy config --debug --quiet=false backup`

so, to avoid these cases, i think we should also have that [check](https://github.com/aquasecurity/trivy/issues/7781#issuecomment-2432560222) if the archive was "removed" in previous recursion steps (just use `fs.Stat` instead of `fs.Glob`, as you suggested)

Reply:
Added a check that the file exists.